packages/core/src/engine/Autodiff.ts

```diff
@@ -277,6 +298,7 @@ const children = (x: VarAD): Child[] => {
 };
 
 const indexToID = (index: number): ad.Id => `_${index}`;
+const idToIndex = (id: ad.Id): number => parseInt(id.slice(1), 10);
```

Comment:
A little weird that you have to parse this. Can't the edge store the indices too?

Reply:
Yeah that makes sense! Just wanted to flag this. Given that I don't really see a better way.